### ISSUE TITLE
ctype? is broken because a high-order value passed as `Any`

### ISSUE THREAD

**Issue by wargrey** [racket/typed-racket#393]
### What version of Racket are you using?
6.6.0.1
(6.5 is okay)

### What program did you run?
https://gist.github.com/wargrey/da46158909267d4b2d6a9efce172bf88

### What should have happened?
display the HASH and HMAC results

### If you got an error message, please include it here.
typed-racket/utils/require-contract.rkt:59:16: ctype?: broke its own contract
  Attempted to use a higher-order value passed as `Any` in untyped code: #<ctype>
  in: the 1st argument of
      a part of the or/c of
      (or/c
       struct-predicate-procedure?/c
       (-> Any boolean?))
  contract from: (interface for ctype?)
  blaming: (interface for ctype?)
   (assuming the contract is correct)

**Comment by AlexKnauth** [comment]
In the very very latest version of typed racket, this error becomes a warning instead. These two commits did that:
https://github.com/racket/typed-racket/commit/50aab9806ff549ddc5f59f00d16ce336bc20eec6
https://github.com/racket/typed-racket/commit/3d80322d272daeeafb3a3211f8a25befad79bc15

**Comment by wargrey** [comment]
Okay, it works.
Although, those warnings are annoying.

**Comment by samth** [comment]
Yes, they are. I have a patch in progress to allow TR to know that `cpointer?` is safe, but in general this isn't an easy problem to solve.

**Comment by wargrey** [comment]
All right, thank you all.
